### PR TITLE
Wording differentiation for DM "chat" instead of "room"

### LINKED ISSUE #2164: Wording differentiation for DM "chat" instead of "room"
### Events formatter for DM "chat"
- [x] "notice_room_created" = "*%@ joined* ~~%@ created the room~~"
- [ ] "notice_room_history_visible_to_anyone" = "N/A ~~%@ made future room history visible to anyone.~~"
- [ ] "notice_room_history_visible_to_members_from_invited_point" = "%@ made future ~~room~~ 
- [ ] *messages* ~~history~~ visible to everyone ~~all room members~~, from **when** ~~the point~~ they **get** ~~are~~ invited."
- [x] "notice_room_third_party_invite" = "%@ invited %@ ~~%@ sent an invitation to %@ to join the room~~"
- [x] "notice_room_update"%s upgraded ~~this room.~~ here. (?)

### Notice Events with "You"
- [x] * "notice_room_third_party_invite_by_you" = "You **invited** ~~sent an invitation to~~ %@ ~~to join the room~~"
- [x] * "notice_room_third_party_revoked_invite_by_you" = "You revoked **%@'s invitation** ~~the invitation for %@ to join the room~~"

### DM Settings
- [x] Room settings = ~~Room~~ Settings
- [x] Leave room = Leave ~~room~~
- [x] Security copy = Messages **here** ~~in this room~~

### Encryption enabled
- [x] Messages here are end-to-end encrypted...

### Others
- [x] "room_left" = "You left ~~the room~~"
- [x] "room_no_power_to_create_conference_call" = "You need permission to invite to start a conference **here** ~~in this room~~"
- [x] "room_no_conference_call_in_encrypted_rooms" = "Conference calls are not supported *when encryption is on* ~~in encrypted rooms~~"
- [ ] "notice_room_history_visible_to_anyone" = "%@ made future ~~room history~~ **messages** visible to anyone."
- [x] "notice_room_third_party_revoked_invite_by_you" = "You revoked **%@'s invitation** ~~the invitation for %@ to join the room~~"
- [x] "notice_room_third_party_invite" = "%@ invited %@ ~~%@ sent an invitation to %@ to join the room~~"
- [x] "notice_room_third_party_invite_by_you" = "You ~~sent an invitation to %@ to join the room~~ invited %@"
- [x] "room_left" = "You left ~~the room~~"

### PR DESCRIPTION
**Pull request by onurays**
Fixes #2164

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt [hunk 1]
```diff
@@ -56,6 +57,27 @@ class NoticeEventFormatter @Inject constructor(private val activeSessionDataSour
...
+        }.orFalse()
+    }
+
+    private fun chooseResourceByRoomType(event: Event,
+                                         @StringRes directRoomResId: Int,
+                                         @StringRes directRoomByUserResId: Int,
+                                         @StringRes publicRoomResId: Int,
+                                         @StringRes publicRoomByUserResId: Int,
+                                         userResArgs: Any? = null,
+                                         vararg thirdPartyResArgs: Any?
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:72]: Please rename those 2 params?

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt [hunk 2]
```diff
@@ -56,6 +57,27 @@ class NoticeEventFormatter @Inject constructor(private val activeSessionDataSour
...
+        return roomId?.let {
+            activeSessionDataSource.currentValue?.orNull()?.getRoomSummary(roomId)?.isDirect
+        }.orFalse()
+    }
+
+    private fun chooseResourceByRoomType(event: Event,
+                                         @StringRes directRoomResId: Int,
+                                         @StringRes directRoomByUserResId: Int,
+                                         @StringRes publicRoomResId: Int,
+                                         @StringRes publicRoomByUserResId: Int,
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:70]: Is not really for public rooms but for not DM rooms. I do not know what to propose though...

File: vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileController.kt [hunk 3]
```diff
@@ -57,7 +57,7 @@ class RoomProfileController @Inject constructor(
         // Security
         buildProfileSection(stringProvider.getString(R.string.room_profile_section_security))
         val learnMoreSubtitle = if (roomSummary.isEncrypted) {
-            R.string.room_profile_encrypted_subtitle
+            if (roomSummary.isDirect) R.string.direct_room_profile_encrypted_subtitle else R.string.room_profile_encrypted_subtitle
         } else {
             R.string.room_profile_not_encrypted_subtitle
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileController.kt:62]: No change for this one?

File: vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileController.kt [hunk 4]
```diff
@@ -71,7 +71,11 @@ class RoomProfileController @Inject constructor(
         buildProfileSection(stringProvider.getString(R.string.room_profile_section_more))
         buildProfileAction(
                 id = "settings",
-                title = stringProvider.getString(R.string.room_profile_section_more_settings),
+                title = if (roomSummary.isDirect) {
+                    stringProvider.getString(R.string.direct_room_profile_section_more_settings)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileController.kt:75]: The test could be done inside the `getString` (at other places too)

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt [hunk 5]
```diff
@@ -56,6 +57,27 @@ class NoticeEventFormatter @Inject constructor(private val activeSessionDataSour
 
     private fun Event.isSentByCurrentUser() = senderId != null && senderId == currentUserId
 
+    private fun isDirectRoom(roomId: String?): Boolean {
+        return roomId?.let {
+            activeSessionDataSource.currentValue?.orNull()?.getRoomSummary(roomId)?.isDirect
+        }.orFalse()
+    }
+
+    private fun chooseResourceByRoomType(event: Event,
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:66]: Rename the method

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt (hunk 2 above)
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:70]: Maybe create a data class with 4 StringRes members for clarity

**Review by bmarty — COMMENTED** [review]
I've made some remarks

File: vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileController.kt (hunk 3 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileController.kt:62]: Good catch.

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt [hunk 6]
```diff
@@ -179,11 +201,14 @@ class NoticeEventFormatter @Inject constructor(private val activeSessionDataSour
...
                 ?.takeIf { it.creator.isNullOrBlank().not() }
                 ?.let {
-                    if (event.isSentByCurrentUser()) {
-                        sp.getString(R.string.notice_room_created_by_you)
-                    } else {
-                        sp.getString(R.string.notice_room_created, it.creator)
-                    }
+                    chooseResourceByRoomType(
+                            event = event,
+                            directRoomResId = R.string.notice_room_created_by_you,
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:206]: This is nopt the correct string res (just as a reminder for me)

**Review by ganfra — APPROVED** [review]
LGTM, thanks

**Pull request merged**